### ISSUE TITLE
bug on prefer destructuring when using with optional chaining

### ISSUE THREAD

**Issue by sibelius** [eslint/eslint#12636]
**Tell us about your environment**

* **ESLint Version:** v6.0.1
* **Node Version:** v12.13.0
* **npm Version:** 6.13.1

**What parser (default, Babel-ESLint, etc.) are you using?**
@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  extends: [
    'airbnb/base',
    'plugin:import/errors',
    'plugin:import/warnings',
    'plugin:react/recommended',
    'plugin:@typescript-eslint/eslint-recommended',
    'plugin:@typescript-eslint/recommended',
    'prettier/@typescript-eslint',
    // 'plugin:prettier/recommended',
  ],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 10,
    sourceType: 'module',
    ecmaFeatures: {
      modules: true,
    },
  },
  rules: {
    quotes: [2, 'single', { allowTemplateLiterals: true, avoidEscape: true }],
    'jsx-quotes': [2, 'prefer-single'],
    'react/prop-types': 0,
    'no-mixed-operators': 0,
    'global-require': 0,
    'no-console': 2,
    'max-len': [1, 120, 2],
    'no-param-reassign': [2, { props: false }],
    'no-continue': 0,
    'no-underscore-dangle': 0,
    'generator-star-spacing': 0,
    'no-duplicate-imports': 0,
    'no-use-before-define': 0,
    'consistent-return': 0,
    'spaced-comment': 0,
    'arrow-parens': 0,
    'import/first': 1,
    'no-return-await': 0,
    'flowtype/define-flow-type': 1,
    'flowtype/use-flow-type': 1,
    'import/namespace': [2, { allowComputed: true }],
    'import/no-duplicates': 1,
    'import/order': [2, { 'newlines-between': 'always-and-inside-groups' }],
    'no-unused-vars': [
      2,
      {
        ignoreRestSiblings: true,
      },
    ],
    'no-await-in-loop': 0,
    'no-plusplus': 0,
    'import/prefer-default-export': 0,
    'no-confusing-arrow': 0,
    'no-restricted-syntax': 0,
    'no-bitwise': 0,
    'function-paren-newline': 0,
    'object-curly-newline': 0,
    'import/no-cycle': 1,
    'import/no-unresolved': [2, { ignore: ['@entria/graphql-mongoose-loader'] }],
    'import/no-self-import': 1,
    'import/extensions': [0, 'never', { ts: 'never' }],
    indent: 0,
    '@typescript-eslint/indent': 0,
    '@typescript-eslint/camelcase': 0,
    '@typescript-eslint/explicit-function-return-type': 0,
    'interface-over-type-literal': 0,
    '@typescript-eslint/consistent-type-definitions': 0,
    'lines-between-class-members': 0,
    '@typescript-eslint/explicit-member-accessibility': 0,
    '@typescript-eslint/prefer-interface': 0,
    '@typescript-eslint/interface-name-prefix': 0,
    '@typescript-eslint/no-non-null-assertion': 1,
    '@typescript-eslint/no-var-requires': 1,
    '@typescript-eslint/no-object-literal-type-assertion': 1,
    'import/no-extraneous-dependencies': 1,
    'prefer-const': [
      2,
      {
        destructuring: 'all',
      },
    ],
    'no-multi-assign': 1,
  },
  plugins: ['flowtype', 'react', 'import', '@typescript-eslint'],
  globals: {
    require: true,
  },
  env: {
    jest: true,
    node: true,
    es6: true,
    browser: true,
  },
  settings: {
    'import/resolver': {
      node: {
        extensions: ['.js', '.jsx', '.ts', '.tsx'],
      },
      'eslint-import-resolver-typescript': true,
      'eslint-import-resolver-lerna': {
        packages: path.resolve(__dirname, 'packages'),
      },
    },
  },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const id = correlations?.id;
```

```bash

```

**What did you expect to happen?**
it should keep like this:
```jsx
const id = correlations?.id;
```

**What actually happened? Please include the actual, raw output from ESLint.**
it changed to

```jsx
const { id } = correlations;
````

causing a bug

**Are you willing to submit a pull request to fix this bug?**

**Comment by kaicataldo** [comment]
ESLint doesn't support stage 3 or below features (you can see our full policy here). Once optional chaining reaches stage 4, the default parser and core rules will be updated to handle the new syntax.

**Comment by sibelius** [comment]
Babel, Flow and Typescript already support this

Any chance to make this not cause bugs?

**Comment by g-plane** [comment]
The 3rd-party parsers support this syntax doesn't mean ESLint will support this, because the default parser only support syntaxes/features at Stage 4.

**Comment by mdjermanovic** [comment]
@sibelius which version of `@typescript-eslint/parser` are you using? It looks like this shouldn't happen with `2.4.0` and above.

**Comment by sibelius** [comment]
updated `@typescript-eslint/parser`

tks
